File: backend/src/schemas/template.py
from pydantic import BaseModel, Field, field_validator
from typing import Optional
from datetime import datetime


class TemplateBase(BaseModel):
    """Base template schema."""
    name: str = Field(..., min_length=1, max_length=255)
    description: Optional[str] = None


class TemplateCreate(TemplateBase):
    """Schema for creating a template."""
    template_type: str = Field(..., pattern="^(word|pdf|image|direct)$")
    markdown_content: str = Field(..., min_length=1)
    original_filename: Optional[str] = None
    original_file_path: Optional[str] = None


class TemplateUpdate(BaseModel):
    """Schema for updating a template."""
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    description: Optional[str] = None
    markdown_content: Optional[str] = Field(None, min_length=1)


class TemplateResponse(TemplateBase):
    """Schema for template response."""
    id: int
    template_type: str
    original_filename: Optional[str]
    original_file_path: Optional[str]
    markdown_content: str
    created_by: Optional[int]
    created_at: datetime
    updated_at: datetime
    is_active: bool

    class Config:
        from_attributes = True


class TemplateListResponse(BaseModel):
    """Schema for paginated template list response."""
    templates: list[TemplateResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class TemplateIdentifiersResponse(BaseModel):
    """Schema for template identifiers response."""
    template_id: int
    identifiers: list[str]


class FileUploadResponse(BaseModel):
    """Schema for file upload response."""
    filename: str
    file_path: str
    markdown_content: str
    message: str

File: frontend/src/types/template.ts
export type TemplateType = 'word' | 'pdf' | 'image' | 'direct'

export interface Template {
  id: number
  name: string
  description: string | null
  template_type: TemplateType
  original_filename: string | null
  original_file_path: string | null
  markdown_content: string
  identifiers: string | null
  created_by: number | null
  created_at: string
  updated_at: string
  is_active: boolean
}

export interface TemplateCreate {
  name: string
  description?: string
  template_type: TemplateType
  markdown_content: string
  original_filename?: string
  original_file_path?: string
}

export interface TemplateUpdate {
  name?: string
  description?: string
  markdown_content?: string
}

export interface TemplateListResponse {
  templates: Template[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface TemplateIdentifiersResponse {
  template_id: number
  identifiers: string[]
}

export interface FileUploadResponse {
  filename: string
  file_path: string
  markdown_content: string
  message: string
}
